### ISSUE TITLE
Only 20 friends loaded from cache

### ISSUE THREAD

**Issue by snakebiter** [skjnldsv/Owncloud-FBSync#15]
Hi. When I click dropdown 'Choose friend' that is located in contact card, I can see only small part (20) of my FB friends. Some of them not showing FB profile photo when choosen.

**Comment by skjnldsv** [comment]
Hi!
Do you have any errors on your log file?
How many friends should it displays?

**Comment by snakebiter** [comment]
OK. I found where the problem is. I'm not fluent in English, and I hope that this is enough to understand me ;-)

In `\OCA\FbSync\Controller\FacebookController::getfriends`
line 180 `$friends[(int)$matches[1]]=html_entity_decode($friend->innertext, ENT_QUOTES);`

You are casting matched FB uid to (int) to use it as array key. The problem is that I have small VPS running under 32-bit Debian. Casting big integer like FB uid to 32-bit integer causes its degradation and many array entries are overwritten.

I have removed casting in my copy of code and everything started working fine. Maybe it's time to think about upgrade to 64 bit system. 

BTW: nice and useful application. Keep going. Best.
